update the vm_build.sh script and add some Vagrantfiles

tesch1 opened the pull request:
- on macos the admin user isnt vnmr1, rather the current user, so allow that.
- added reference "gold" vjqa files for different OSs to track support state.
- updated build_vm.sh to print a summary at the end of a (potentially very long) run.
- updated vjqa "gold" references to reflect current builds
- added mint19 and fedora28 Vagrantfiles

DanIverson commented:
I tried ./build_vm.sh mint19 and got
unrecognized arg: mint19
I added mint19 to build_vm.sh in the places it seemed to be missing and tried again. This time, I got
./build_vm.sh: line 152: /space/ovjbuild/ovjTools/bin/build_box.sh: No such file or directory
Setting up build VM in /space/ovjbuild/ovjTools/vms/mint19
==> default: VM not created. Moving on...

DanIverson commented:
I tried build_vm.sh and got some errors. I realized that I was using an old version. I cloned a new ovjTools and checked out your changes. Now I get
./build_vm.sh mint19
INFO :Logfile: /space/ovjbuild/ovjTools/logs/build_vm-mint19-20190302_093407.log
INFO :Target:  mint19
INFO :Actions: 
WARN : Target 'mint19' wasn't started, skipping shutdown.
INFO : Finished with 'mint19'
INFO :
INFO :build_vm.sh done, 0 errors.  Logfile: /space/ovjbuild/ovjTools/logs/build_vm-mint19-20190302_093407.log

I tried ./build_vm.sh mint19 clean and got
INFO :Logfile: /space/ovjbuild/ovjTools/logs/build_vm-mint19-20190302_093716.log
INFO :Target:  mint19
INFO :Actions:  clean
INFO : -----------------------------------------
WARN :     ACTION=clean TARGET=mint19
INFO : -----------------------------------------
INFO :Cleaning vms/mint19
INFO :Cmd started Sat Mar  2 09:37:16 PST 2019
INFO :$ vagrant destroy -f
.
INFO :Cmd returned: 0, duration: 
build_vm.sh: Error on line 620, exiting.

DanIverson commented:
If I supply the "build" action, it starts building the VM but eventually fails.
varian:bin vnmrjvarian$ ./build_vm.sh mint19 build
INFO :Logfile: /space/ovjbuild/ovjTools/logs/build_vm-mint19-20190302_094952.log
INFO :Target:  mint19
INFO :Actions:  build
INFO : -----------------------------------------
WARN :     Setup 'mint19'
INFO :Setting up target 'mint19'...
INFO :Cmd started Sat Mar  2 09:49:52 PST 2019
INFO :$ vagrant halt
.
INFO :Cmd returned: 0, duration: 
INFO :Spinning up build VM in /space/ovjbuild/ovjTools/vms/mint19
INFO :Cmd started Sat Mar  2 09:49:56 PST 2019
INFO :$ vagrant up
.
INFO :Cmd returned: 1, duration: 
ERROR:Unable to start 'mint19'
WARN : Target 'mint19' wasn't started, skipping shutdown.
INFO : Finished with 'mint19'
INFO :
INFO :build_vm.sh done, 1 errors.  Logfile: /space/ovjbuild/ovjTools/logs/build_vm-mint19-20190302_094952.log

When I checked the VirtualBox managers, there was a mint19_default<number> running. I am doing this on a MacOS High Sierra host PC.

tesch1 commented:
first though - is `vagrant` available?
I'll try it on my mac now...

tesch1 commented:
(just added a couple minor commits - when the branch is good i'll rebase it down to one commit.)

DanIverson commented:
Yes, vagrant is installed. I also installed the plugin with the command
vagrant plugin install vagrant-vbguest

tesch1 commented:
hm, looks like a problem with `printf` on mac being different from unix... working on it..

DanIverson commented:
I was running VirtualBox 5.1.30. I updated to version 5.2.26. It still fails. Also, is it possible to control the location of the .vagrant.d directory? It is about 2 GB in my home directory.  Is it possible to put it in the "ovjbuild" directory, which on my system is a large USB drive?

tesch1 commented:
Ifound this:
https://stackoverflow.com/questions/14733681/vagrant-d-outside-of-the-home-folder
Looks like you can set VAGRANT_HOME to tell it where to put .vagrant.d,
cant' think of any reason it cant be in ovjbuild.
You might need to 'vagrant destroy' the old vms though if you move it - if
there are any hard-coded paths.
Only the macos10.10 vagrant build does anything funky with mounting drives
in the VM that might make this not work, but since you're on macos it's
probably a waste to try to boot a macos vm.

I'm still waiting for the mint19 image to download.


On Sat, Mar 2, 2019 at 7:27 PM Dan Iverson <notifications@github.com> wrote:

> I was running VirtualBox 5.1.30. I updated to version 5.2.26. It still
> fails. Also, is it possible to control the location of the .vagrant.d
> directory? It is about 2 GB in my home directory. Is it possible to put it
> in the "ovjbuild" directory, which on my system is a large USB drive?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/OpenVnmrJ/ovjTools/pull/41#issuecomment-468946837>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AATsL8qrMYzobbR3HHTJrvyR3X3wC-ilks5vSsKrgaJpZM4bYi8L>
> .
>

DanIverson commented:
It seems to be a problem with VirtualBox. I remember using VBoxHeadless sometimes failed. Is it possible to configure vagrant to start the VM in "gui" mode?

tesch1 commented:
Yep, just set the environment variable VMGUI=y (see the Vagrantfile for
this mechanism)
VMGUI=y ./build_vm.sh mint19 build
I suppose it would be convenient to make it a command-line option too.

On Sun, Mar 3, 2019 at 12:34 AM Dan Iverson <notifications@github.com>
wrote:

> It seems to be a problem with VirtualBox. I remember using VBoxHeadless
> sometimes failed. Is it possible to configure vagrant to start the VM in
> "gui" mode?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/OpenVnmrJ/ovjTools/pull/41#issuecomment-468970821>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AATsL6VPRpTQlG5II71RwhINO3RaGuvsks5vSwqRgaJpZM4bYi8L>
> .
>

tesch1 commented:
Seems that Vagrant doesn't like the ethernet driver emulator from vbox selected on that particular Vagrant box because it apparently has some sort of security issue.  Fair enough, I dont really know anything about the reliability of that mint image, no reason to trust it, so let's use a different driver.  The latest commit switches the driver, and fixes more printf usage.  'build' was successful for me, I'm running 'install' and 'test' now.  Under the linux VM, 'test' takes about 14 hours, similar to ubuntu 18 and fedora 28.

DanIverson commented:
I pulled your latest changes. The printf is working. I did vagrant mint19 clean and it worked. I wanted to start from a clean slate so I thought distclean would do that. It failed.

varian:bin vnmrjvarian$ ./build_vm.sh mint19 distclean
Setting OVJ_TOOLS to '/Users/vnmrjvarian/ovjbuild/ovjTools'
INFO :$ vagrant version
Installed Version: 2.2.4
Latest Version: 2.2.4
 
You're running an up-to-date version of Vagrant!
INFO :Logfile: /Users/vnmrjvarian/ovjbuild/ovjTools/logs/build_vm-mint19-20190304_081447.log
INFO :Target:  mint19
INFO :Actions:  distclean
INFO : -----------------------------------------
WARN :     ACTION=distclean TARGET=mint19
INFO : -----------------------------------------
INFO :Cleaning vms/mint19
INFO :Cmd started Mon Mar  4 08:14:49 PST 2019
INFO :$ vagrant destroy -f
.
INFO :Cmd returned: 0, duration: 00:00:04
INFO :distcleaning vms/box_defs/mint19
WARN :target box def for 'mint19' missing
ERROR:distclean 'mint19' failed. (00:00:04) Vagrant VM at /Users/vnmrjvarian/ovjbuild/ovjTools/vms/mint19
WARN : Target 'mint19' wasn't started, skipping shutdown.
INFO : Finished with 'mint19'
INFO :
distclean 'mint19' failed. (00:00:04): /Users/vnmrjvarian/ovjbuild/ovjTools/logs/build_vm-mint19-20190304_081447.log
INFO :vagrant global-status after cleaning:
INFO :$ vagrant global-status

I manually removed the ~/.vagrant.d/boxes/mrl...

./build_vm.sh mint19 in headless mode again failed with "ERROR:Unable to start 'mint19'. I then did
export VMGUI=y
./build_vm.sh mint19 downloaded and imported the OS. It was in the process of installing additional packages (avahi-daemon, avaahi-utils,, muffin-dbg, muffin) when my entire Mac froze with the "beachball". The freeze is repeatable. It may be that vagrant is giving too many resources to the VM. It gave it 3 processors. My Mac only has 2. It gave it 3072 MB 0f Base Memory, leaving only 1024 MB for my host MacOS.  That could be the problem. I usually never give a VM more than 2048 MB.

tesch1 commented:
Ok, confirmed that works for me on macos 10.13.6.  VirtualBox 5.2.26.  Any luck on your end?

```
clean 'mint19' success. (00:00:07)
build 'mint19' success. (01:15:17)
install 'mint19' success. (00:03:16)
test 'mint19' success. (15:06:09)
```

tesch1 commented:
Added some code to check the actual number of cores available, and give half to the VM.  Makes the assumption that everyone (who's doing a ovj build in a VM) has a multi-core machine.

tesch1 commented:
Also, the -N switch to build_vm.sh will override this, you could try rerunning with `-N 1`.  The Vagrantfile is setup to give the VM 1G memory per core.  I thought that was a reasonable approximation, though the hard-coded 3 cores was perhaps not.

DanIverson commented:
The build_vm.sh mint19 worked. Great. When I tried build_vm.sh mint19 distclean, it still failed. I am assuming that will remove files from .vagrant.d/boxes. Is it related to the fact that mint19 is missing from ovjTools/vms/box_defs? fedora28 and ubuntu18 are also missing.  I also noticed that the vjqa step was very slow. I have some ideas why that would be but it is easier to troubleshoot on a CentOS system. When I tried build_vm.sh centos6, it built a VM with 8 GB of memory and 4 processors. My mac only has 4 GB of memory and 2 processors.

DanIverson commented:
When I tried to use
  build_vm.sh centos6 -N 2
it built a centos-6.10... (base) VM with 1 GB of memory and 1 processor. It then apparently cloned that one and made centos6_default which has 8 GB of memory and 4 processors.

tesch1 commented:
Probably was the other way around - the centos6 box had 4/8192 hardwired.  The "boxes" aren't strictly necessary, they are much faster when you want to build from a fresh state, but after a while I felt it's not worth the extra time of setting them up for builds that I'm not doing very often and dont need to prove to myself that the build is good from scratch; so just adding a directory under `vms/` with a Vagrantfile should be enough to add a new platform target.  (`distclean` now ignores targets with no box def).

I've added `-M|--vmmem` and `-N|--ncpu` for memory and ncpus, respectively.  Also `-G|--vmgui` to enable the GUI easily.  And cleaned up and unified all the Vagrantfiles to use this same interface (passing these args through env vars).

DanIverson commented:
All your new feature work correctly. The only error I got for centos6 was:

INFO :$ scp -F vagrant.ssh.config default:~vnmr1/vnmrsys/ovj_qa/ovjtest/reports/report.txt /Users/vnmrjvarian/ovjbuild/ovjTools/logs/vjqa-centos6-20190306_065434.txt
WARN :vjqa results differ from the 'gold' for 'centos6' on       20 lines...
WARN :  diff /Users/vnmrjvarian/ovjbuild/ovjTools/logs/gold_vjqa/vjqa-centos6.txt /Users/vnmrjvarian/ovjbuild/ovjTools/logs/vjqa-centos6-20190306_065434.txt
INFO :$ diff /Users/vnmrjvarian/ovjbuild/ovjTools/logs/gold_vjqa/vjqa-centos6.txt /Users/vnmrjvarian/ovjbuild/ovjTools/logs/vjqa-centos6-20190306_065434.txt | head
/Users/vnmrjvarian/ovjbuild/ovjTools/bin/loglib.sh: line 98: diff /Users/vnmrjvarian/ovjbuild/ovjTools/logs/gold_vjqa/vjqa-centos6.txt /Users/vnmrjvarian/ovjbuild/ovjTools/logs/vjqa-centos6-20190306_065434.txt | head: No such file or directory
MSG  :test 'centos6' success. (00:59:56)

Afterward, when I manually did the diff, it worked.
I also tried the macos10.10 build. I think it failed because I ran out of disk space. It wanted to download a huge file for over 2 hours. When it tried to unpack it, I think that is when I ran out of space. Is there a way to tell vagrant to download stuff in some other location before unpacking it in .vagrant.d. Otherwise, this might be a good excuse for me to buy a new Mac.

tesch1 commented:
Hm, I'm not sure why that diff failed, I dont see any obvious bug, and it worked when I did exactly the same (centos6) yesterday.  Although I got 12 different lines instead of 20 :

```
$ diff /Volumes/Open/OpenVnmrJBuild/ovjTools/logs/gold_vjqa/vjqa-centos6.txt /Volumes/Open/OpenVnmrJBuild/ovjTools/logs/vjqa-centos6-20190306_104556.txt
4c4
< Feb 23 2019 at 11:16:44
---
> Mar 06 2019 at 11:37:14
8c8
< February 23, 2019
---
> March 06, 2019
181c181
<         Passed: 6398 xml files
---
>         Passed: 6395 xml files
```

(maybe the timestamp isn't necessary to have in the file? then could really define a simple pass/fail)

the interwebs seems to claim that VAGRANT_HOME is also the location that vagrant downloads into, so that might be an option.  But getting a new machine is never the wrong decision!  I got a pretty decent new dell box for < 1000 last year, nearly the same price as an equivalent build-your-own, but none of the mucking about.  (european prices: your mileage may vary, favorably).

DanIverson commented:
One possibility would be to delete the first 10 lines from the "gold" files. That would remove the date and vnmrrev information. If a diff between the gold copy and the build copy had more than 10 lines different, that could mean a fail, otherwise pass. I consistently get the diff failure. Subsequent manual diff always works. Very curious.

tesch1 commented:
1eb3d5e should fix it.  Now the only mystery is why the error wasn't showing up for me.

DanIverson commented:
Your changes fixed the diff issue. When I build a mint19 VM, sometimes it works and sometimes it doesn't. The relevant sections of the log files are below.
The bad one shows:
==> default: Booting VM...
==> default: Waiting for machine to boot. This may take a few minutes...
    default: SSH address: 127.0.0.1:2222
    default: SSH username: vagrant
    default: SSH auth method: private key
    default: Warning: Remote connection disconnect. Retrying...
    default: Warning: Remote connection disconnect. Retrying...
    default: Warning: Remote connection disconnect. Retrying...
    default: Warning: Remote connection disconnect. Retrying...
    default: Warning: Remote connection disconnect. Retrying...
    default: Warning: Remote connection disconnect. Retrying...
    default: Warning: Remote connection disconnect. Retrying...
    default: Warning: Remote connection disconnect. Retrying...
    default: Warning: Remote connection disconnect. Retrying...
    default: Warning: Remote connection disconnect. Retrying...
    default: Warning: Remote connection disconnect. Retrying...
    default: Warning: Remote connection disconnect. Retrying...
    default: Warning: Remote connection disconnect. Retrying...
    default: Warning: Remote connection disconnect. Retrying...
    default: Warning: Remote connection disconnect. Retrying...
    default: Warning: Remote connection disconnect. Retrying...
    default: Warning: Remote connection disconnect. Retrying...
    default: Warning: Remote connection disconnect. Retrying...
    default: Warning: Remote connection disconnect. Retrying...
Timed out while waiting for the machine to boot. This means that
Vagrant was unable to communicate with the guest machine within
the configured ("config.vm.boot_timeout" value) time period.

If you look above, you should be able to see the error(s) that
Vagrant had when attempting to connect to the machine. These errors
are usually good hints as to what may be wrong.

If you're using a custom box, make sure that networking is properly
working and you're able to connect to the machine. It is a common
problem that networking isn't setup properly in these boxes.
Verify that authentication configurations are also setup properly,
as well.

If the box appears to be booting properly, you may want to increase
the timeout ("config.vm.boot_timeout") value.

The good one shows:

==> default: Booting VM...
==> default: Waiting for machine to boot. This may take a few minutes...
    default: SSH address: 127.0.0.1:2222
    default: SSH username: vagrant
    default: SSH auth method: private key
    default: Warning: Remote connection disconnect. Retrying...
    default: Warning: Remote connection disconnect. Retrying...
    default: Warning: Remote connection disconnect. Retrying...
    default: Warning: Remote connection disconnect. Retrying...
    default: Warning: Remote connection disconnect. Retrying...
    default: Warning: Remote connection disconnect. Retrying...
    default: Warning: Remote connection disconnect. Retrying...
    default: Warning: Remote connection disconnect. Retrying...
    default: Warning: Remote connection disconnect. Retrying...
    default: Warning: Remote connection disconnect. Retrying...
    default: Warning: Remote connection disconnect. Retrying...
    default: Warning: Remote connection disconnect. Retrying...
    default: Warning: Remote connection disconnect. Retrying...
    default: Warning: Remote connection disconnect. Retrying...
    default: Warning: Remote connection disconnect. Retrying...
    default: Warning: Remote connection disconnect. Retrying...
    default: Warning: Remote connection disconnect. Retrying...
    default: Warning: Remote connection disconnect. Retrying...
    default: Warning: Remote connection disconnect. Retrying...
    default:
    default: Vagrant insecure key detected. Vagrant will automatically replace
    default: this with a newly generated keypair for better security.
    default:
    default: Inserting generated public key within guest...
    default: Removing insecure key from the guest if it's present...
    default: Key inserted! Disconnecting and reconnecting using new SSH key...
==> default: Machine booted and ready!

tesch1 commented:
I guess the root question there is whether the machine is still booting because you machine is slow, or if this particular mint19 image is hanging for some reason?  as far as I remember, I havne't seen it hang. What happens if you add `config.vm.boot_timeout = 800` to the Vagrantfile?

DanIverson commented:
In both the good and bad cases, it displays the 
  default: Warning: Remote connection disconnect. Retrying...
message 19 times. The VM is running and windows are displayed before the sixth "Retrying..." message is displayed. It is another 2 minutes until the last "Retrying..." message is displayed. It does not sound like a timeout issue.

DanIverson commented:
In a few days, I am going to push changes that fix a lot of the vjqa warnings.

I tried ubuntu18 and it failed.  The section of the log file is:
    default: Setting up python3-distupgrade (1:18.04.30) ...
    default: Setting up python3-update-manager (1:18.04.11.9) ...
    default: Setting up ubuntu-release-upgrader-core (1:18.04.30) ...
    default: 
    default: Configuration file '/etc/update-manager/release-upgrades'
    default:  ==> Modified (by you or by a script) since installation.
    default:  ==> Package distributor has shipped an updated version.
    default:    What would you like to do about it ?  Your options are:
    default:     Y or I  : install the package maintainer's version
    default: 
    default:     N or O  : keep your currently-installed version
    default:       D     : show the differences between the versions
    default:       Z     : start a shell to examine the situation
    default:  The default action is to keep your current version.
    default: *** release-upgrades (Y/I/N/O/D/Z) [default=N] ? 
    default: Configuration file '/etc/update-manager/release-upgrades'
    default:  ==> Modified (by you or by a script) since installation.
    default: 
    default:  ==> Package distributor has shipped an updated version.
    default:    What would you like to do about it ?  Your options are:
    default:     Y or I  : install the package maintainer's version
    default: 
    default:     N or O  : keep your currently-installed version
    default:       D     : show the differences between the versions
    default:       Z     : start a shell to examine the situation
    default: 
    default:  The default action is to keep your current version.
    default: *** release-upgrades (Y/I/N/O/D/Z) [default=N] ? dpkg: error processing package ubuntu-release-upgrader-core (--configure):
    default:  end of file on stdin at conffile prompt
    default: dpkg: dependency problems prevent configuration of update-manager-core:
    default:  update-manager-core depends on ubuntu-release-upgrader-core (>= 1:18.04.9); however:
    default:   Package ubuntu-release-upgrader-core is not configured yet.
    default: 
    default: dpkg: error processing package update-manager-core (--configure):
    default:  dependency problems - leaving unconfigured
    default: dpkg: dependency problems prevent configuration of update-notifier-common:
    default:  update-notifier-common depends on update-manager-core (>= 1:17.04.2); however:
    default:   Package update-manager-core is not configured yet.
    default: 
    default: dpkg: error processing package update-notifier-common (--configure):
    default:  dependency problems - leaving unconfigured
    default: Processing triggers for libc-bin (2.27-3ubuntu1) ...
    default: No apport report written because the error message indicates its a followup error from a previous failure.
    default: No apport report written because the error message indicates its a followup error from a previous failure.
    default: Processing triggers for initramfs-tools (0.130ubuntu3.7) ...
    default: update-initramfs: Generating /boot/initrd.img-4.15.0-29-generic
    default: Errors were encountered while processing:
    default:  ubuntu-release-upgrader-core
    default:  update-manager-core
    default:  update-notifier-common
    default: E
    default: : 
    default: Sub-process /usr/bin/dpkg returned an error code (1)
The SSH command responded with a non-zero exit status. Vagrant
assumes that this means the command failed. The output for this command
should be in the log above. Please read the output to determine what
went wrong.
INFO :Cmd returned: 1, duration: 00:10:08
ERROR:Unable to start 'ubuntu18'
WARN : Target 'ubuntu18' wasn't started, skipping shutdown.
INFO : Finished with 'ubuntu18'
INFO :
clean 'ubuntu18' success. (00:00:12)
INFO :build_vm.sh done, 1 errors.  Logfile: /Users/vnmrjvarian/ovjbuild/ovjTools/logs/build_vm-ubuntu18-20190311_095448.log

tesch1 commented:
Hm, when I have a little more time I'll see if I can find a better mint19 image - I wasn't really excited about the one I did find, it's just from some random guy, not the mint project nor vagrant nor bento.  I haven't tried the ubuntu18 for a while, I'll have a look in the next few days, that should be easier to fix.

DanIverson commented:
The ubuntu14 and ubuntu16 builds both worked correctly, except the ubutbu16 did not have "zip" installed. I am not sure if ubuntu14 and ubuntu18 have zip installed. The unzip command was installed on ubuntu16. The fedora28 build gave the following error
==> default: Checking for guest additions in VM...
==> default: Running provisioner: shell...
    default: Running: inline script
    default: Last metadata expiration check: 0:06:34 ago on Tue 12 Mar 2019 11:58:48 PM UTC.
    default: No match for argument: epel-release
    default: Error: Unable to find a match
The SSH command responded with a non-zero exit status. Vagrant
assumes that this means the command failed. The output for this command
should be in the log above. Please read the output to determine what
went wrong.
INFO :Cmd returned: 1, duration: 00:13:42
ERROR:Unable to start 'fedora28'
WARN : Target 'fedora28' wasn't started, skipping shutdown.
INFO : Finished with 'fedora28'

The scons process in src/vjqa leaves several files in OpenVnmrJ/src/vjqa with root ownership. It would be nice to fix that if it is not too much trouble.

Also, I pushed some changes to fix a lot of the vjqa warnings for psglib and some for the ldd tests. That may change the gold-circleci file.

tesch1 commented:
added some more updates, but they're untested - running the builds over the day today.

Is there still a machine in oregon for us?  Maybe we could setup nightly/weekly builds on all the VMs?

DanIverson commented:
Your new stuff now works with ubuntu18. I also tried ubuntu18 on the Oregon machine but it failed. The log file is below. The printf failure is because the Oregon machine is running RH 6.9 with bash version 4.1. The %(fmt)T feature was not introduced until bash version 4.2. I am not sure what caused the other failure.

INFO :Cmd started Wed Mar 13 06:56:40 PDT 2019
INFO :$ vagrant up
Bringing machine 'default' up with 'virtualbox' provider...
==> default: Box 'bento/ubuntu-18.04' could not be found. Attempting to find and install...
    default: Box Provider: virtualbox
    default: Box Version: 201812.27.0
The box 'bento/ubuntu-18.04' could not be found or
could not be accessed in the remote catalog. If this is a private
box on HashiCorp's Atlas, please verify you're logged in via
`vagrant login`. Also, please double-check the name. The expanded
URL and error message are shown below:

URL: ["https://atlas.hashicorp.com/bento/ubuntu-18.04"]
Error: The requested URL returned error: 404 Not Found
/home/diverson/ovjbuild/ovjTools/bin/loglib.sh: line 106: printf: `(': invalid format character
INFO :Cmd returned: 1, duration: 
ERROR:Unable to start 'ubuntu18'
WARN : Target 'ubuntu18' wasn't started, skipping shutdown.
INFO : Finished with 'ubuntu18'

DanIverson commented:
The mint19 seems to be unpredictable. The first time it failed after the 19 "Retrying..." messages.  However, rather than the windows in the VM show up before the sixth Retrying message, none showed up until build_vm exits with the error about networking. As soon as the script ended, the windows showed up in the VM.

The next three times, the VM window showed
  "FATAL: Could not read from the boot medium! System halted."

I removed the mint image from .vagrant.d/boxes and the next time it worked.

The next time it failed with the networking problem.
Then it worked again.

The fedora29 builds the VM but the ovj build fails. There is a missing lib.nsl.so.1 library, which is needed by the java categories.

tesch1 commented:
I've changed the mint19 box source and built (+installed +tested) this new one four times (3 with linux host, once under macos) with no problems, hopefully it works for you too?

tesch1 commented:
Fedora 29 support is currently blocked by these changes (removal of long-deprecated rpc from glibc):
https://fedoraproject.org/wiki/Changes/NISIPv6
https://fedoraproject.org/wiki/Changes/SunRPCRemoval

I'm experimenting with using libtirpc instead of glibc's on the branch https://github.com/OpenVnmrJ/OpenVnmrJ/pull/256

DanIverson commented:
I built mint19 three times. It worked each time. Perfect.

DanIverson commented:
Are you planning more changes or is this ready to be merged?

tesch1 commented:
I was thinking to wait for the rpc stuff to be resolved, but I guess that's not necessary.  Let me run it again against ovj/master and update all the "gold" vjqa files and then rebase everything... should be done tomorrow...

tesch1 commented:
now I remember where I was at... did you want me to remove 'infoproc', 'nvinfoproc' and 'stat' from ovj?  Sounded like they're not used anymore?  I dont know exactly what they're for - is it correct to think that if someone ever wanted to replace their old sun/vnmr they might be needed?

I've been building against my messy `fix-bashloc` branch in order to get fedora28 and ubuntu18 to build, but as you may have correctly guessed - I dont have any way to test the change to using tirpc.

tesch1 commented:
Here's what I propose... I'll remove all the tirpc stuff from this PR, it will break the newer linuxes that have removed it from glibc, but will get this PR through with support for all the other target OSs : building OVJ/master for centos6/7 ubuntu14/16 and mint19.  We can deal with the rpc stuff separately, later.

DanIverson commented:
I had made some comments in OpenVnmrJ/OpenVnmrJ#256

tesch1 commented:
Yep, just saw those.

tesch1 commented:
okie dokey, against the current OpenVnmrJ/master:

distclean 'ubuntu18' success. (00:00:01)
build 'ubuntu18' success. (00:35:22)
install 'ubuntu18' success. (00:01:08)
test 'ubuntu18' success. (00:47:40)
distclean 'centos6' success. (00:00:03)
build 'centos6' success. (00:38:29)
install 'centos6' success. (00:00:53)
test 'centos6' success. (00:35:28)
distclean 'centos7' success. (00:00:02)
build 'centos7' success. (01:01:14)
install 'centos7' success. (00:01:07)
test 'centos7' success. (00:38:34)
distclean 'mint19' success. (00:00:00)
build 'mint19' success. (00:44:51)
install 'mint19' success. (00:00:49)
test 'mint19' success. (00:47:53)
distclean 'ubuntu14' success. (00:00:02)
build 'ubuntu14' success. (00:29:09)
install 'ubuntu14' success. (00:00:22)
test 'ubuntu14' success. (00:42:33)
distclean 'ubuntu16' success. (00:00:03)
build 'ubuntu16' success. (00:38:58)
install 'ubuntu16' success. (00:00:59)
test 'ubuntu16' success. (00:37:32)

DanIverson commented:
I installed your latest changes. I cannot build macos10.10. I run out of disk space during the 4 hr. download. I really do need to get a new computer. When I tried fedora29, it imported the VM and installed the packages. Then, I got the following errors.

==> default: VM started with 2 cpus, 2048 MB memory, and gui=true
^[[32mINFO :Cmd returned: 0, duration: 00:44:41^[(B^[[m
^[[35mfedora29$ echo "export ovjBuildDir=$HOME/ovjbuild" >> ~/.bashrc^[(B^[[m
^[[37m^[[1mFAILED=255 00:03:06^[(B^[[m
^[[35mfedora29$ echo "export OVJ_TOOLS=$ovjBuildDir/ovjTools" >> ~/.bashrc^[(B^[[m
Connection to 127.0.0.1 closed.^M
^[[35mOK 00:00:23^[(B^[[m
^[[35mfedora29$ echo "export OVJ_NMRADMIN=vnmr1" >> ~/.bashrc^[(B^[[m
Connection to 127.0.0.1 closed.^M
^[[35mOK 00:00:05^[(B^[[m
^[[35mfedora29$ echo ovjBuildDir=$ovjBuildDir^[(B^[[m
ovjBuildDir=^M
Connection to 127.0.0.1 closed.^M
^[[35mOK 00:00:05^[(B^[[m
^[[35mfedora29$ echo OVJ_TOOLS=$OVJ_TOOLS^[(B^[[m
OVJ_TOOLS=/ovjTools^M
Connection to 127.0.0.1 closed.^M
^[[35mOK 00:00:04^[(B^[[m
^[[35mfedora29$ scons -v^[(B^[[m
SCons by Steven Knight et al.:^M
        script: v3.0.1.74b2c53bc42290e911b334a6b44f187da698a668, 2017/11/14 13:16:53, by bdbaddog on hpmicrodog^M
        engine: v3.0.1.74b2c53bc42290e911b334a6b44f187da698a668, 2017/11/14 13:16:53, by bdbaddog on hpmicrodog^M
        engine path: ['/usr/lib/python2.7/site-packages/SCons']^M
Copyright (c) 2001 - 2017 The SCons Foundation^M
Connection to 127.0.0.1 closed.^M
^[[35mOK 00:00:07^[(B^[[m
^[[32mINFO : -----------------------------------------^[(B^[[m
^[[33mWARN :     ACTION=build TARGET=fedora29^[(B^[[m
^[[32mINFO : -----------------------------------------^[(B^[[m
^[[32mINFO :validating repositories for OpenVnmrJ and ovjTools...^[(B^[[m
^[[32mINFO :http://github.com/OpenVnmrJ/OpenVnmrJ branch=master valid^[(B^[[m
^[[32mINFO :http://github.com/OpenVnmrJ/ovjTool branch=master is valid^[(B^[[m
^[[32mINFO :Building target VM 'fedora29'...^[(B^[[m
^[[35mfedora29$ echo OVJ_TOOLS=$OVJ_TOOLS^[(B^[[m
OVJ_TOOLS=/ovjTools^M
Connection to 127.0.0.1 closed.^M
^[[35mOK 00:00:05^[(B^[[m
^[[35mfedora29$ mkdir -p ${ovjBuildDir}^[(B^[[m
mkdir: missing operand^M
Try 'mkdir --help' for more information.^M
Connection to 127.0.0.1 closed.^M
^[[37m^[[1mFAILED=1 00:00:04^[(B^[[m
^[[31mERROR:build 'fedora29' failed. (00:48:49) Vagrant VM at /Users/vnmrjvarian/ovjbuild/ovjTools/vms/fedora29^[(B^[[m

tesch1 commented:
Strange.  I just ran fedora29 again with no problems (other than the expected failure at rpc), but I just noticed that you had the GUI enabled, so I'm trying that now.  Could you have run out of memory on the fedora build?  I guess I'd expect a different error in that case, but maybe it errored earlier silently? could you attach the whole log file?

tesch1 commented:
It also "works" (rtc excepted) for me under linux with `-G`: Vagrant 2.1.2, VirtualBox 5.2.26.  Just looked closer at your message... this looks like the root of the problem:
```
 ^[[35mfedora29$ echo "export ovjBuildDir=$HOME/ovjbuild" >> ~/.bashrc^[(B^[[m
^[[37m^[[1mFAILED=255 00:03:06^[(B^[[m
```
On osx this fedora29 also failed for me twice - one at the same line, but the VM just hung instead of having that command fail.  The second time it got past that just fine, but hung later randomly during compilation.  Feels like maybe a bug in VirtualBox on osx, triggered by this box?

DanIverson commented:
I put the entire log file on dropbox. You can get it here https://www.dropbox.com/s/hpahaq0ictis98q/mt.zip?dl=0
When you are ready, I will merge the changes.

tesch1 commented:
Hm, well I still don't know.  It's failing on the first time it tries to ssh to the machine, one would imagine that after the installation of all the packages and `vagrant up` returns that the machine would be booted and ready to go, but maybe not... I've added a non-critical "echo hello world" to "prime" the VM & ssh.. maybe that will do it.  I'm ready to merge.  If it's still broken in the future we'll fix it then.

tesch1 commented:
@DanIverson Is there anything else left before merging this?
